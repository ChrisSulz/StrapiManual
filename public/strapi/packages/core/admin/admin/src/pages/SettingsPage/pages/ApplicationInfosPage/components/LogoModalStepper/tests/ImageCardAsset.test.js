import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render as renderTL } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import ImageCardAsset from '../ImageCardAsset';

const ASSET_FIXTURES = {
  ext: '.jpeg',
  height: 250,
  name: 'asset.jpeg',
  size: 46.26,
  url: 'uploads/asset.jpeg',
  width: 340,
};

const render = (props) =>
  renderTL(
    <ThemeProvider theme={lightTheme}>
      <IntlProvider locale="en" messages={{}} textComponent="span">
        <ImageCardAsset asset={ASSET_FIXTURES} {...props} />
      </IntlProvider>
    </ThemeProvider>
  );

describe('ApplicationInfosPage | ImageCardAsset', () => {
  it('should render and match snapshot', () => {
    const { container } = render();

    expect(container).toMatchSnapshot();
  });

  it('should display and format asset extension and size', () => {
    const { getAllByText, getByText } = render();

    getAllByText(/\.jpeg/i).map((element) => expect(element).toBeInTheDocument());
    expect(getByText(/340âœ•250/i)).toBeInTheDocument();
  });
});
