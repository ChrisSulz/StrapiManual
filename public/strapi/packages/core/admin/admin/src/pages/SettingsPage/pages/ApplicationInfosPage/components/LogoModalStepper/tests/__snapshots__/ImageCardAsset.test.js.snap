// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationInfosPage | ImageCardAsset should render and match snapshot 1`] = `
.c17 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c9 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c10 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c16 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c0 {
  background: #ffffff;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c1 {
  position: relative;
}

.c6 {
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
}

.c12 {
  background: #eaeaef;
  padding-right: 8px;
  padding-left: 8px;
  min-width: 20px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  border-radius: 4px;
  height: 1.5rem;
}

.c5 {
  margin: 0;
  padding: 0;
  max-height: 100%;
  max-width: 100%;
  object-fit: contain;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.5rem;
  width: 100%;
  background: repeating-conic-gradient(#f6f6f9 0% 25%,transparent 0% 50%) 50% / 20px 20px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c11 {
  margin-left: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c15 {
  margin-left: 4px;
}

.c8 {
  word-break: break-all;
}

.c3 {
  border-bottom: 1px solid #eaeaef;
}

<div>
  <article
    aria-labelledby=":r0:-title"
    class="c0"
    tabindex="0"
  >
    <div
      class="c1 c2 c3"
    >
      <div
        class="c4"
      >
        <img
          aria-hidden="true"
          class="c5"
          src="uploads/asset.jpeg"
        />
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <div
            class="c9"
            id=":r0:-title"
          >
            asset.jpeg
          </div>
          <div
            class="c10"
          >
            .JPEG - 340âœ•250
          </div>
        </div>
        <div
          class="c11"
        >
          <div
            class="c12 c13 c14 c15"
          >
            <span
              class="c16"
            >
              image
            </span>
          </div>
        </div>
      </div>
    </div>
  </article>
  <div
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
